File: pattern.cpp
 //    1
 //   121
 //  12321
 // 1234321  git init
// git config user.name "someone"
// git config user.email "someone@someplace.com"
// git add *
// git commit -m "some init msg

#include <iostream>
using namespace std;





int main(){ 


   int n=4;
   int i=0;
   int x=0;
   int space=n;
   


   for(int row=0;row<n;row++){ 

     space--;
     int col=2*row+1;
    
    for(int k=0;k<space;k++){ 
     cout<<" ";

    }
    for(int j=0;j<col;j++){ 
    
    if(j>row){ 
      x--;
      cout<<x;
     
    }else{ 
      x++;
     cout<<x;
    
    }


    }
    x=0;

    cout<<endl;



   }



//   return 0;

//    }


/////////////////
///////////////
//
 //    *
 //   * *
 //  * * *
 // * * * *

// int main(){ 
// 	int n=5;
// 	int row=n;
// 	int col=0;
// 	int space=n;
	
// 	for(int i=0;i<row;i++){ 
//     col=i;
      
//       for(int j=0;j<space;j++){ 
//         cout<<" ";
//       }
//       space--;
//       for(int k=0;k<col;k++){ 
//        cout<<"*"<<" ";
//       }
//       cout<<endl;



// 	}
// }


////////////
////////
  

//            *     
//          * *
//        * * *
//      * * * * 
//    * * * * * 

// int main(){ 


// int n =5;
// int space=5;
// for(int row=0;row<n;row++){ 


//    for(int s=0;s<space;s++){ 
//     cout<<" ";
//    }
//    space--;
//    for(int col=0;col<=row;col++){ 

//       cout<<"*";
//    }
//    cout<<endl;




// }
// return 0;


// }

//2nd method for same pattern using only 2 loops

// int main(){ 


//     int n=5;
//     int x=0;

//     for(int row=1;row<=n;row++){ 

//      int var=n;
     
//      for(int col=1;col<=n;col++){ 
      
//        if((var-col)>x){ 
//       cout<<" ";

//        }else{ 
//        cout<<"*";
//        }
        

//      }
//      cout<<endl;
//      x++;


//     }









//    return 0;
// }


////3rd method by decrementing space by 2

int main(){ 
   
int n=5;
int space=(2*n)-2;

for(int row=0;row<n;row++){ 

 for(int s=0;s<space;s++){ 
     cout<<" ";
 }
 space=space-2;

 for(int col=0;col<=row;col++){ 
    cout<<"* ";
 }
 cout<<endl;




}





return 0;
}

File: pattern.py
#    1
#   121
#  12321
# 1234321
##progrm for above pattern
# n=4
# space=n
# row=n
# x=0
# for i in range(0,row):
#   space=space-1
#   col=2*i+1
#   for j in range(0,space):
#     print(end=" ")
#   for k in range(0,col):
#     if(k>i):
#       x=x-1
#       print(x,end="")
#     else:
#       x=x+1
#       print(x,end="")
#   x=0 
#   print('\r')

########
######

 #    *
 #   * *
 #  * * *
 # * * * *

 ##program for above program 


# n=4
# space=n
# for i in range(0,n):
#   for s in range(0,space):
#     print(end=" ")
#   space=space-1
#   for col in range(0,i+1):
#     print("*",end=" ")

#   print('\r')



###
######
######

#           *     
#         * *
#       * * *
#     * * * * 
#   * * * * * 
# n=4
# space=n
# for row in range(0,n):
#   for s in range(0,space):
#     print(end=" ")
#   space=space-1
#   for col in range(0,row+1):
#     print("*",end="")
#   print("\r")

#####
####
#### 2nd method for this using only 2 loops
# n=5
# var=n-1
# x=0
# for row in range(0,n):
# 	for col in range(0,n):
# 		if(var-col>x):
# 			print(end=" ")
# 		else:
# 			print("*",end="")
# 	print("\r")
# 	x=x+1

###another method bu decrementing space by 2
n=5
space=2*n-2
for row in range(0,n):
	for s in range(0,space):
		print(end=" ")
	space=space-2
	for col in range(0,row+1):
		print("* `",end="")
	print("\r")
	